components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/ContextLoader.java

```diff
@@ -88,24 +97,56 @@ public static URI buildURIForBody(String endpoint) {
      */
     public static URI buildURIForHeader(String endpoint) {
 
-        String tenantQualifiedRelativePath =
-                String.format(TENANT_CONTEXT_PATH_COMPONENT, getTenantDomainFromContext()) + SERVER_API_PATH_COMPONENT;
-        String url = IdentityUtil.getEndpointURIPath(tenantQualifiedRelativePath + endpoint, false, true);
-
-        URI loc = URI.create(url);
-        if (!loc.isAbsolute()) {
-            Message currentMessage = PhaseInterceptorChain.getCurrentMessage();
-            if (currentMessage != null) {
-                UriInfo ui = new UriInfoImpl(currentMessage.getExchange().getInMessage(), null);
-                try {
-                    return new URI(ui.getBaseUri().getScheme(), ui.getBaseUri().getAuthority(), url, null, null);
-                } catch (URISyntaxException e) {
-                    LOG.error("Server encountered an error while building the location URL with scheme: " +
-                            ui.getBaseUri().getScheme() + ", authority: " + ui.getBaseUri().getAuthority() +
-                            ", url: " + url, e);
-                }
-            }
+        URI loc;
+        String context = getContext(endpoint);
+
+        try {
+            String url = ServiceURLBuilder.create().addPath(context).build().getAbsolutePublicURL();
+            loc = URI.create(url);
+        } catch (URLBuilderException e) {
+            String errorDescription = "Server encountered an error while building URL for response header.";
+            throw buildInternalServerError(e, errorDescription);
         }
         return loc;
     }
+
+    /**
+     * This method builds the API context by checking whether the tenant qualified url is enabled or not.
```

Comment:
Better to explain why we need this method.
ie. in tenant qualified mode ServiceURLBuilder appends the tenant domain as a path param automatically. But in not tenant qualified mode it doesn't. So we need to append the tenant domain manually in the path